test/CreatorToken.t.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {Test, console2} from "forge-std/Test.sol";
+import {CreatorToken} from "src/CreatorToken.sol";
+import {IERC20, ERC20} from "openzeppelin/token/ERC20/ERC20.sol";
+
+contract CreatorTokenTest is Test {
+  IERC20 public payToken;
+  CreatorToken public creatorToken;
+  address public creator = address(0xc2ea702);
+  address public admin = address(0xb055);
+
+  string CREATOR_TOKEN_NAME = "Test Token";
+  string CREATOR_TOKEN_SYMBOL = "TEST";
+
+  string PAY_TOKEN_NAME = "Payment Token";
+  string PAY_TOKEN_SYMBOL = "PAY";
+  uint256 BASE_PAY_AMOUNT = 1e18; // Because our test token has 18 decimals
+
+  event Bought(
+    address indexed _payer,
+    address indexed _receiver,
+    uint256 indexed _tokenId,
+    uint256 _paymentAmount,
+    uint256 _creatorFee,
+    uint256 _adminFee
+  );
+
+  function setUp() public {
+    payToken = new ERC20(PAY_TOKEN_NAME, PAY_TOKEN_SYMBOL);
+    creatorToken =
+      new CreatorToken(CREATOR_TOKEN_NAME, CREATOR_TOKEN_SYMBOL, creator, admin, payToken);
+  }
+
+  // TODO: consider, should this be a view method on the contract itself?
+  function calculateTotalPrice(uint256 _price) public view returns (uint256) {
+    (uint256 _creatorFee, uint256 _adminFee) = creatorToken.calculateFees(_price);
+    return _price + _creatorFee + _adminFee;
+  }
+}
+
+contract Deployment is CreatorTokenTest {
+  function test_TokenIsConfiguredAtDeployment() public {
+    assertEq(creatorToken.name(), CREATOR_TOKEN_NAME);
+    assertEq(creatorToken.symbol(), CREATOR_TOKEN_SYMBOL);
+    assertEq(creatorToken.creator(), creator);
+    assertEq(creatorToken.admin(), admin);
+    assertEq(address(creatorToken.payToken()), address(payToken));
+  }
+
+  function test_FirstTokenIsMintedToCreator() public {
+    assertEq(creatorToken.balanceOf(creator), 1);
+    assertEq(creatorToken.ownerOf(1), creator);
+  }
+}
+
+contract Buying is CreatorTokenTest {
+  function test_SecondTokenIsBoughtForOnePaymentToken(address _buyer) public {
+    vm.assume(_buyer != address(0) && _buyer != address(creatorToken));
+    uint256 originalBuyerBalance = creatorToken.balanceOf(_buyer);
+
+    (uint256 _creatorFee, uint256 _adminFee) = creatorToken.calculateFees(BASE_PAY_AMOUNT);
+    uint256 _totalPrice = BASE_PAY_AMOUNT + _creatorFee + _adminFee;
+    deal(address(payToken), _buyer, _totalPrice);
+
+    vm.startPrank(_buyer);
+    payToken.approve(address(creatorToken), type(uint256).max);
+    creatorToken.buy(_totalPrice);
+    vm.stopPrank();
+
+    assertEq(creatorToken.balanceOf(_buyer), originalBuyerBalance + 1);
+    assertEq(payToken.balanceOf(_buyer), 0);
```

Comment:
`assertEq(payToken.balanceOf(_buyer), 0);` is failing when `_buyer` is the `creator` address that receives 7% fee. We can add `_buyer != creator` to `vm.assume`.

Reply:
should also exclude `admin`.
All this also applies to another test: `test_SecondTokenIsBoughtForOnePaymentToken`

Reply:
These changes are made in the following commits of https://github.com/showtime-xyz/creator-tokens/pull/9
https://github.com/showtime-xyz/creator-tokens/pull/9/commits/5c25d5f9dfc0f115a3f8c28dfd3253469ef0f9ec
https://github.com/showtime-xyz/creator-tokens/pull/9/commits/70428f5fc967283d6c5657d2dae122f62c3f3776

Reply:
Nice catch and fix @garyghayrat–looks like we have another failure though. Mind investigating that one as well?

Reply:
Those changes were made in a different PR. But since you agree on the approach, I've made the changes to this PR in this commit https://github.com/showtime-xyz/creator-tokens/pull/4/commits/e3bb4847237e44c9da441a31a0a670d9e7ae594b and CI is passing now.